### PR TITLE
Add markdown readme for System.Reflection.MetadataLoadContext

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Issue: https://github.com/dotnet/runtime/issues/59630

Previous related PR: https://github.com/dotnet/runtime/pull/69015

I've decided to include example here, unlike in previous PR, because MLC does not have a lot of conceptual docs with examples on Microsoft Docs site, and also its API surface is not evolving very fast, so it won't be much hard to maintain the additional example. I can remove it if consistency is more important.

### CODE REVIEW

File: src/libraries/System.Reflection.MetadataLoadContext/src/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+- [How to: Inspect assembly contents using MetadataLoadContext](https://docs.microsoft.com/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext)
+- [System.Reflection.MetadataLoadContext](https://docs.microsoft.com/dotnet/api/system.reflection.metadataloadcontext)
+- [System.Reflection.MetadataAssemblyResolver](https://docs.microsoft.com/dotnet/api/system.reflection.metadataassemblyresolver)
+
+## Example
+
+The following example shows how print the list of types defined in an assembly.
+
+```
```
Comment (buyaa-n) [inline src/libraries/System.Reflection.MetadataLoadContext/src/README.md:15]: NIT: Not sure if adding `cs` would work for md file and make the code look better
```suggestion
```cs
```

**Review by buyaa-n — APPROVED** [review]
Overall LGTM, thanks @MSDN-WhiteKnight!

File: src/libraries/System.Reflection.MetadataLoadContext/src/README.md (hunk 1 above)
Comment (MSDN-WhiteKnight) [reply src/libraries/System.Reflection.MetadataLoadContext/src/README.md:15]: Syntax highlighting is currently not supported in NuGet (https://github.com/NuGet/NuGetGallery/issues/8632). But it wouldn't hurt to add language indicator so syntax highlighting will light up on Github or in some third-party viewers that support it.

**Comment by buyaa-n** [comment]
Failure related to https://github.com/dotnet/runtime/issues/70856

**Pull request merged**